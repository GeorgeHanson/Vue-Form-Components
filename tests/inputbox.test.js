import Vue from 'vue'
import InputBoxComponent from '../src/Components/InputBox.vue';

describe('InputBoxComponent', () => {
    test("it generates the markup", () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            render: (createElement) => {
                return createElement(
                    InputBoxComponent,
                    {
                        props: {
                            name: 'name',
                            label: 'Name'
                        }
                    }
                )
            },
        });

        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});