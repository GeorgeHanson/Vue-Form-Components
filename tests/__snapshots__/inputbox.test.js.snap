// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBoxComponent it generates the markup 1`] = `
< div server - rendered = "true"
class = "form-group" > < label class = "control-label col-sm-2" > Name: <!----></label> <div class="col-sm-10"><input type="text" name="name" class="form-control"> <!----></div> <div class="clearfix"></div></div>
`;
